woocommerce/woocommerce-android
# Unified Login: Track events for WPcom login flow

Closes #2721 by adding the final tracks events for the WPcom login flow. This includes:
- Setting the flow to `wordpress_com` when the `Continue with WordPress.com` button is clicked. 
- Track events for google login
- Persisting `source` and `flow` track state to `AppPrefs` so if the user closes the app on the store picker screen and returns later we can restore those two values. 

**Note**: Track events for magic link and password were already added in a previous PR

### Sample Track events

**Flow: Wordpress.com + magic link**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"continue_with_wordpress_com"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"enter_email_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"request_magic_link"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"magic_link_requested"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"open_email_client"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"request_magic_link"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"magic_link_requested"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"open_email_client"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"success"}
```

**Flow: WordPress.com + password**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"continue_with_wordpress_com"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"enter_email_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"login_with_password"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_password","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_password","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_password","step":"success"}
```

**Flow: Google Login (2FA)**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"continue_with_wordpress_com"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"enter_email_address"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"google_login","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit_2fa_code"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"success"}
```

### To Test
- Test logging in with WordPress.com flow w/ magic link flow
- Test logging in with WordPress.com flow w/ password flow
- Test logging in with WordPress.com flow w/ google login
- Test closing the app on the store picker screen and restoring the app. The `source` and `flow` should be reinstated and the `success` event tracked successfully. For example: 
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"success"}
```


Update release notes:
